src/main/java/com/google/api/gax/grpc/AbstractPage.java

```diff
@@ -107,42 +112,75 @@ public RequestT getRequest() {
     return request;
   }
 
-  /** Package-private for use by PagedListResponseContext */
+  public int getPageElementCount() {
+    return Iterables.size(pageDescriptor.extractResources(response));
+  }
+
+  public abstract AbstractPage<RequestT, ResponseT, ResourceT> getNextPage(int pageSize);
+
+  protected <PageT> PageT getNextPage(PageFactory<RequestT, ResponseT, ResourceT, PageT> provider) {
+    if (hasNextPage()) {
+      RequestT nextRequest = pageDescriptor.injectToken(request, getNextPageToken());
+      return callApiAndCreate(provider, callable, pageDescriptor, nextRequest, context);
```

Comment:
I think that actually a PageContext would help, have added. See other comments also.